packages/salesforcedx-vscode-core/src/commands/forceApexLogFetch.ts

```diff
@@ -0,0 +1,182 @@
+    const resultJson = JSON.parse(result);
+    if (resultJson.status === 0) {
+      const logDir = path.join(
+        vscode.workspace.workspaceFolders![0].uri.fsPath,
+        '.sfdx',
+        'tools',
+        'debug',
+        'logs'
+      );
+      if (!fs.existsSync(logDir)) {
+        mkdir('-p', logDir);
+      }
+      const date = new Date(response.data.startTime)
+        .toISOString()
+        .replace('T', ' ')
+        .replace('.000', '');
+      fs.writeFile(
+        path.join(logDir, `${response.data.id}_${date}.log`),
+        resultJson.result.log
+      );
+    }
+  }
+}
+
+export enum ApexDebugLogRequest {
+  Api = 'Api',
+  Application = 'Application'
+}
+
+export type ApexDebugLogIdStartTime = {
+  id: string;
+  startTime: string;
+};
+
+export type ApexDebugLogObject = {
+  Id: string;
+  StartTime: string;
+  LogLength: number;
+  Operation: string;
+  Request: string;
+};
+
+export async function getLogs(
+  cancellationTokenSource: vscode.CancellationTokenSource
+): Promise<ApexDebugLogObject[]> {
+  const execution = new CliCommandExecutor(
+    new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_apex_log_list_text'))
+      .withArg('force:apex:log:list')
+      .withArg('--json')
+      .build(),
+    { cwd: vscode.workspace.workspaceFolders![0].uri.fsPath }
+  ).execute();
+  CancellableStatusBar.show(execution, cancellationTokenSource);
+  taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  const cmdOutput = new CommandOutput();
+  const result = await cmdOutput.getCmdResult(execution);
+  try {
+    const apexDebugLogObjects: ApexDebugLogObject[] = JSON.parse(result).result;
+    apexDebugLogObjects.sort(function(a, b): number {
+      return new Date(b.StartTime).valueOf() - new Date(a.StartTime).valueOf();
+    });
+    return Promise.resolve(apexDebugLogObjects);
+  } catch (e) {
+    return Promise.reject(e);
+  }
+}
+
+export interface ApexDebugLogItem extends vscode.QuickPickItem {
+  id: string;
+}
+
+export class LogFileSelector
+  implements ParametersGatherer<ApexDebugLogIdStartTime> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<ApexDebugLogIdStartTime>
+  > {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const logInfos = await getLogs(cancellationTokenSource);
+    if (logInfos.length > 0) {
+      const logItems = logInfos.map(logInfo => {
+        let icon = '$(alert) ';
+        switch (logInfo.Request) {
+          case ApexDebugLogRequest.Api:
+            icon = '$(rocket) ';
+            break;
+          case ApexDebugLogRequest.Application:
+            icon = '$(squirrel) ';
+            break;
+        }
+        return {
+          id: logInfo.Id,
+          label: icon + logInfo.Operation,
+          description: logInfo.StartTime,
+          detail: String(logInfo.LogLength)
+        } as ApexDebugLogItem;
+      });
+      const logItem = await vscode.window.showQuickPick(
```

Comment:
Passed in the same cancellation token for getLog so this isn't displayed if the getLog command is cancelled